packages/block-library/src/post-author-name/index.php

```diff
@@ -26,6 +26,10 @@ function render_block_core_post_author_name( $attributes, $content, $block ) {
 		return '';
 	}
 
+	if ( ! post_type_supports( get_post_type( $block->context['postId'] ), 'author' ) ) {
```

Comment:
```suggestion
	if ( ! post_type_supports( $block->context['postType'], 'author' ) ) {
```

We can directly get the post type via context as long as it's listed in the `usesContext` array in the `block.json`